x/crosschain/keeper/evm_hooks.go

```diff
@@ -356,12 +349,43 @@ func (k Keeper) validateZRC20Withdrawal(
 				constant.SolanaWalletRentExempt,
 			)
 		}
+	}
+
+	// Validate the destination address
+	return k.validateDestinationAddress(chainID, to, additionalChains)
+}
+
+// validateCall validates the data of a gateway Call event
+// it checks that destination address is supported depending on the chain
+func (k Keeper) validateCall(
+	ctx sdk.Context,
+	chainID int64,
+	to []byte,
+) error {
+	additionalChains := k.GetAuthorityKeeper().GetAdditionalChainList(ctx)
+	return k.validateDestinationAddress(chainID, to, additionalChains)
+}
+
+// validateDestinationAddress validates the destination address based on the chain type
+func (k Keeper) validateDestinationAddress(
```

Comment:
Why creating this sub function? in validateZRC20Withdrawal we already have logic dispatched per chain, then we dispatch again here

It would be cleaner to me to have all verification logic per chain in a single dispatch. Eventually having a sub fucntion per chain

Reply:
Ok, I understand it's because we have `validateCall`

I would suggest rather to refactor validateZRC20Withdrawal and make it more general like validateOutbound and it can be used in both workflow